pkg/project/v2/project_loader.go

```diff
@@ -191,22 +189,46 @@ func loadProject(fs afero.Fs, context ProjectLoaderContext, projectDefinition ma
 		}
 	}
 
-	// The scope parameter of a key user actions web configuration needs to be a reference to another application-web config
-	// The reference parameter makes sure that rely on the fact that kua web configs are loaded/deployed within the same
-	// sub graph (independent component) later on as
 	for _, c := range configs {
+		// The scope parameter of a key user actions web configuration needs to be a reference to another application-web config
+		// The reference parameter makes sure that rely on the fact that kua web configs are loaded/deployed within the same
+		// sub graph (independent component) later on as
 		if c.Coordinate.Type == api.KeyUserActionsWeb {
 			if _, ok := c.Parameters[config.ScopeParameter].(*ref.ReferenceParameter); !ok {
 				errors = append(errors, fmt.Errorf("scope parameter of config of type '%s' with ID '%s' needs to be a reference "+
 					"parameter to another web-application config", api.KeyUserActionsWeb, c.Coordinate.ConfigId))
 			}
 		}
 	}
-
 	if errors != nil {
 		return Project{}, errors
 	}
 
+	// Network zone API configs have a hard dependency on the "builtin:networkzones" setting 2.0 config
+	// i.e. if a network zone AND a "builtin:networkzones" config is going to be deployed, we need to make sure the
+	// "builtin:networkzones" config comes first, hence we insert a fixed parameter between the networkzones API configs and
+	// the single "builtin:networkzones" config
+	var networkZones []config.Config
+	var networkZoneEnabled config.Config
+	var networkZoneEnabledFound bool
+	for _, c := range configs {
+		if c.Coordinate.Type == api.NetworkZone {
+			networkZones = append(networkZones, c)
+		}
+		if c.Coordinate.Type == "builtin:networkzones" {
+			networkZoneEnabled = c
+			networkZoneEnabledFound = true
+		}
+	}
+	// if we've found "networkzones" and "builtin:networkzones" settings 2.0 objects, then insert a fixed parameter.
```

Comment:
nit: this comment feels somewhat redundant, you have extensive description before the block, and that this loop adds the ref param is IMO clear from code.